packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -351,166 +380,213 @@ export interface EmbeddedChunk<ChunkMetadata extends Jsonifiable = Jsonifiable>
 
 /** A chunk along with its score for a particular query. */
 export interface ScoredChunk<ChunkMetadata extends Jsonifiable = Jsonifiable> {
-  chunk: EmbeddedChunk<ChunkMetadata>;
+  /** The Chunk result. */
+  chunk: Chunk<ChunkMetadata>;
+  /** The Chunk's raw score for the given query. */
   score: number;
+
+  /** The id of the vector encoding this chunk (if supported). */
+  id?: string;
 }
 
 /**
  * A corpus is a collection of documents that can be provided to an LLM to provide grounding for
- * responses, along with the machinery to load and index those documents.
+ * responses.
  */
 export interface Corpus<ChunkMetadata extends Jsonifiable = Jsonifiable> {
-  /**
-   * Begins loading documents into the corpus using a Loader, Chunker, and Embedding.
-   *
-   * Note: This returns a promise because some implementations may make an external request
-   * to begin loading. However, the promise will typically resolve prior to the corpus being fully
-   * loaded. In such cases, use `getStats` to check progress.
-   */
-  startLoading: () => Promise<Corpus.Stats>;
-
-  /**
-   * Reveals statistics about this corpus, including its LoadingState.
-   *
-   * Note: This returns a promise because some implementations may make an external request.
-   */
-  getStats: () => Promise<Corpus.Stats>;
-
-  /**
-   * Finds document chunks that are semantically similar to the provided query.
-   */
+  /** Finds document chunks that are semantically similar to the provided query. */
   search: (
     query: string,
     params?: { limit?: number; score_threshold?: number }
   ) => Promise<ScoredChunk<ChunkMetadata>[]>;
 }
 
-export namespace Corpus {
-  /** An indication of what steps are necessary to make this corpus ready for serving. */
-  export enum LoadingState {
-    /** Documents have not been loaded. Call `startLoading` to begin loading them. */
-    NOT_STARTED = 'NOT_STARTED',
-    /** Documents are currently being loaded. */
-    IN_PROGRESS = 'IN_PROGRESS',
-    /** Documents were successfully loaded. The corpus is ready for serving. */
-    COMPLETED = 'COMPLETED',
-    /** Documents failed to load. See details for your implementation to try to resolve. */
-    FAILED = 'FAILED',
-  }
+/** An indication of what steps are necessary to make this corpus ready for serving. */
+export enum CorpusLoadingState {
+  /** Documents have not been loaded. Call `startLoading` to begin loading them. */
+  NOT_STARTED = 'NOT_STARTED',
+  /** Documents are currently being loaded. */
+  IN_PROGRESS = 'IN_PROGRESS',
+  /** Documents were successfully loaded. The corpus is ready for serving. */
+  COMPLETED = 'COMPLETED',
+  /** Documents failed to load. See details for your implementation to try to resolve. */
```

Comment:
We may wish to link to https://docs.ai-jsx.com/guides/observability as well.

Reply:
I'm not sure I understand what you're getting at here.  Are you just wanting to point out where logs could be?